recipes/qt/5.x.x/conanfile.py

```diff
@@ -737,7 +740,7 @@ def build(self):
 
         libdirs = [l for dependency in self.dependencies.host.values() for l in dependency.cpp_info.aggregated_components().libdirs]
         args.append("QMAKE_LIBDIR+=\"%s\"" % " ".join(libdirs))
-        if not is_msvc(self):
+        if not is_msvc(self) and not self._is_clang():
```

Comment:
Is this a problem on newer versions of Clang or only on version 13? If it's only an issue for 13, I think we should only do this for that version of Clang. I think we want to keep RPATH enabled otherwise.

Reply:
well, I cannot really tell, I can confirm that the issue exist for clang 13 and 14 on Macos m1

Reply:
I beleive this is an issue with the linker and not the compiler. The linker in Apple platforms doesn't have `-rpath-link`, but on Linux (even with Clang), this should work.


Perhaps this conditional can be changed to `if not is_msvc(self) and not is_apple_os(self):` - or something along those lines.

Reply:
probably `if not is_msvc(self) and not (is_apple_os(self) and self._is_clang()) :` is more accurate, assuming `-rpath-link` is available for gcc on macos